src/test/mod.rs

```diff
@@ -1478,3 +1478,36 @@ fn test_deglob() {
         ],
     );
 }
+
+#[test]
+fn test_all_targets() {
+    let mut env = Environment::new("bin_lib");
+
+    let root_path = env.cache.abs_path(Path::new("."));
+
+    let messages = vec![
+        initialize(0, root_path.as_os_str().to_str().map(|x| x.to_owned())).to_string(),
+    ];
+
+    env.with_config(|c| {
+        c.all_targets = true;
+        c.cfg_test = true;
+    });
+    let (mut server, results) = env.mock_server(messages);
+    // Initialize and build.
+    assert_eq!(
+        ls_server::LsService::handle_message(&mut server),
+        ls_server::ServerStateChange::Continue
+    );
+    expect_messages(
+        results.clone(),
+        &[
+            ExpectedMessage::new(Some(0)).expect_contains("capabilities"),
+            ExpectedMessage::new(None).expect_contains("beginBuild"),
+            ExpectedMessage::new(None).expect_contains("diagnosticsBegin"),
+            // warning from integration tests - test_data/bin_lib/tests/tests.rs
+            ExpectedMessage::new(None).expect_contains("unused variable: `unused_var`"),
```

Comment:
Could you add the path to the expectation so we can't pass if there is an unused variable in the bin or lib